crates/sui-adapter/src/programmable_transactions/types.rs

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use move_binary_format::file_format::AbilitySet;
+use move_core_types::language_storage::StructTag;
+use move_vm_types::loaded_data::runtime_types::Type;
+use serde::Deserialize;
+use sui_types::{
+    base_types::SuiAddress,
+    coin::Coin,
+    error::{ExecutionError, ExecutionErrorKind},
+    object::{Data, MoveObject, Object, Owner},
+};
+
+pub enum Value {
+    Object(ObjectValue),
+    Raw(ValueType, Vec<u8>),
+}
+
+pub struct ObjectValue {
+    pub type_: StructTag,
+    pub has_public_transfer: bool,
+    // None for objects created this transaction
+    pub owner: Option<Owner>,
+    // true if it has been used in a public Move call
+    // In other words, false if all usages have been with non-Move comamnds or
+    // entry Move functions
+    pub used_in_public_move_call: bool,
+    pub contents: ObjectContents,
+}
+
+pub enum ObjectContents {
+    Coin(Coin),
+    Raw(Vec<u8>),
+}
+
+pub enum ValueType {
+    Any,
+    Loaded { ty: Type, abilities: AbilitySet },
+}
+
+impl Value {}
```

Comment:
spurious?

Reply:
Keeping around as they got filled in above, but I'll remove them at the end if they end up being empty